File: android/src/main/java/com/skjline/rideraid/ant/AntDevice.java
package com.skjline.rideraid.ant;

import android.app.Activity;

public interface AntDevice {
    int ANT_DEVICE_ACTIVE = 0xFF;
    int ANT_DEVICE_ERROR = -1;

    int ANT_DEVICE_TYPE_CADENCE = 0;
    int ANT_DEVICE_TYPE_SPEED = 1;

    int getType();
    void release();

    boolean isActive();
    void activate(final Activity activity);
}

File: rider/src/main/java/com/pss9/rideraid/ant/AntDevice.kt
package com.pss9.rideraid.ant

import android.app.Activity

interface AntDevice {

    fun release()
    fun activate(activity: Activity)

    companion object {
        const val ANT_DEVICE_ACTIVE = 0xFF
        const val ANT_DEVICE_ERROR = -1

        const val ANT_DEVICE_TYPE_CADENCE = 0
        const val ANT_DEVICE_TYPE_SPEED = 1
    }
}
